### DISCUSSION TITLE
APKTool Error!

### DISCUSSION THREAD

**Discussion by johnconner122** [ReVanced/revanced-discussions#901] [Question] [answered]
Is there a solution to this error: 
`WARNING: Cant find 9patch chunk in file: "v/ak8.9.png". Renaming it to *.png.`
`WARNING: Cant find 9patch chunk in file: "ag/ang.9.png". Renaming it to *.png.`

I was trying to decompile `com.snaptube.premium` and can't recompile again!!

**Comment by oSumAtrIX** [comment]
ReVanced won't touch resources if no resource patch is included.

**Comment by johnconner122** [comment]
It wasn't revanced specific. I decompiled apk with and recompiling it:
`java.exe -jar apktool_2.7.0.jar b -f --use-aapt2 -o compiled.apk`


I: Building resources...
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\animator\a.xml:4: error: attribute state_liftable (aka com.snaptube.premium:state_liftable) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\animator\a.xml:4: error: attribute state_lifted (aka com.snaptube.premium:state_lifted) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\animator\r.xml:4: error: attribute state_dragged (aka com.snaptube.premium:state_dragged) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\color\e6.xml:4: error: attribute state_highLight (aka com.snaptube.premium:state_highLight) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\color\e6.xml:5: error: attribute state_unHighLight (aka com.snaptube.premium:state_unHighLight) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\color\e7.xml:4: error: attribute state_highLight (aka com.snaptube.premium:state_highLight) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\color\e7.xml:5: error: attribute state_unHighLight (aka com.snaptube.premium:state_unHighLight) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\color\wv.xml:5: error: attribute state_dragged (aka com.snaptube.premium:state_dragged) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\color\wv.xml:6: error: attribute state_dragged (aka com.snaptube.premium:state_dragged) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\color\ww.xml:5: error: attribute state_dragged (aka com.snaptube.premium:state_dragged) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\drawable\fh.xml:4: error: attribute state_warning (aka com.snaptube.premium:state_warning) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\drawable\fh.xml:5: error: attribute state_highLight (aka com.snaptube.premium:state_highLight) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\drawable\fh.xml:6: error: attribute state_unHighLight (aka com.snaptube.premium:state_unHighLight) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\layout\a.xml:9: error: attribute horizontalSpacing (aka com.snaptube.premium:horizontalSpacing) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\layout\a.xml:9: error: attribute verticalSpacing (aka com.snaptube.premium:verticalSpacing) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\layout\a08.xml:5: error: attribute rippleColor (aka com.snaptube.premium:rippleColor) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\layout\a0f.xml:4: error: attribute layout_constraintStart_toStartOf (aka com.snaptube.premium:layout_constraintStart_toStartOf) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\layout\a0f.xml:4: error: attribute layout_constraintTop_toTopOf (aka com.snaptube.premium:layout_constraintTop_toTopOf) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\layout\a0f.xml:5: error: attribute layout_constraintStart_toStartOf (aka com.snaptube.premium:layout_constraintStart_toStartOf) not found.
W: D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\res\layout\a0f.xml:5: error: attribute layout_constraintTop_toBottomOf (aka com.snaptube.premium:layout_constraintTop_toBottomOf) not found.
brut.androlib.AndrolibException: brut.common.BrutException: could not exec (exit code = 1): [C:\Users\Rizwan\AppData\Local\Temp\brut_util_Jar_137187984001730602615726191859812105267.tmp, link, -o, C:\Users\Rizwan\AppData\Local\Temp\APKTOOL16434991549774183423.tmp, --package-id, 127, --min-sdk-version, 21, --target-sdk-version, 23, --version-code, 6194710, --version-name, 6.19.0.6194710, --no-auto-version, --no-version-vectors, --no-version-transitions, --no-resource-deduping, --allow-reserved-package-id, -e, C:\Users\Rizwan\AppData\Local\Temp\APKTOOL17533325130611610324.tmp, -0, arsc, -I, C:\Users\Rizwan\AppData\Local\apktool\framework\1.apk, --manifest, D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\AndroidManifest.xml, D:\Softwares\Apps & Softwares\Android\Android Apps\Snaptube [6.19.0.6194710]\build\resources.zip]
Compile failed. Please read the log

**Comment by oSumAtrIX** [comment]
>  It wasn't revanced specific. 

In that case open an issue in the relevant org/repository. Decoding resources is failing for androlib

**Reply by johnconner122** [reply, marked as answer]
Solved using aapt, instead of aapt2

**Reply by oSumAtrIX** [reply]
Can you check if aapt also works for YouTube?

**Reply by johnconner122** [reply]
Yes, it works.

**Reply by oSumAtrIX** [reply]
Sounds good thanks.
